transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java

```diff
@@ -0,0 +1,238 @@
+    Schema inputSchema = stageConfigurer.getInputSchema();
+    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
+    config.validateBinaryExecutableType();
+    config.validateInputFields(inputSchema);
+    config.validateOutputFieldType();
+    stageConfigurer.setOutputSchema(buildOutputSchema(inputSchema));
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    if (!Strings.isNullOrEmpty(config.fieldsToProcess)) {
+      for (String inputField : Splitter.on(',').trimResults().split(config.fieldsToProcess)) {
+        inputFieldsToProcess.add(inputField);
+      }
+    }
+    executor = new RunExternalProgramExecutor(config.commandToExecute);
+    executor.startAndWait();
+
+  }
+
+  @Override
+  public void transform(StructuredRecord structuredRecord, Emitter<StructuredRecord> emitter) throws Exception {
+    outputSchema = outputSchema == null ? buildOutputSchema(structuredRecord.getSchema()) : outputSchema;
+    StringBuilder executableArguments = new StringBuilder();
+
+    for (String inputField : inputFieldsToProcess) {
+      Schema inputFieldSchema = structuredRecord.getSchema().getField(inputField).getSchema();
+      Schema.Type inputFieldType = inputFieldSchema.isNullable() ? inputFieldSchema.getNonNullable().getType() :
+        inputFieldSchema.getType();
+
+      if ((inputFieldType.equals(Schema.Type.STRING)) &&
+        ((String) (structuredRecord.get(inputField))).contains(" ")) {
+        executableArguments.append("\"");
+        executableArguments.append(structuredRecord.get(inputField));
+        executableArguments.append("\"");
+      } else {
+        executableArguments.append(structuredRecord.get(inputField));
+      }
+      executableArguments.append(" ");
+    }
+
+    // append the fixed arguments in the command, if any
+    if (!Strings.isNullOrEmpty(config.fixedArguments)) {
+      executableArguments.append(config.fixedArguments);
+    }
+
+    executor.submit(executableArguments.toString().trim(), emitter, structuredRecord, outputSchema);
+  }
+
+  @Override
+  public void destroy() {
+    executor.stopAndWait();
+  }
+
+  /**
+   * Builds the emitter's final output schema using the output field provided, along with the input fields.
+   *
+   * @param schema
+   * @return output schema
+   */
+  private Schema buildOutputSchema(Schema schema) {
+    List<Schema.Field> fields = new ArrayList<>(schema.getFields());
+    // Since binary to be executed, can produce null the output for a particular input, hence creating nullable schema
+    // for the output field
+    Schema.Field field = Schema.Field.of(config.outputField, Schema.nullableOf(Schema.of(Schema.Type.valueOf(
+      config.outputFieldType.trim().toUpperCase()))));
+    if (fields.contains(field)) {
+      throw new IllegalArgumentException(String.format("Field %s already has type specified. Duplicate field %s found.",
```

Comment:
IMO we should throw error and that too only if the field already present is not nullable. User should be able to add the field to configuration as well as output schema.